### PR TITLE
Examples of external generators moved to O2DPG (get rid of AEGIS)

### PR DESCRIPTION
**Pull request by preghenella**
hello @shahor02 
with this PR I start to get ready to get rid of the `AEGIS` requirement in the `O2` and move macros for external generators in the `O2DPG` repository.

These files are moved in `O2DPG` by the following commits
https://github.com/AliceO2Group/O2DPG/commit/b1f0bde8b992caa25086587ecb09aaab44b50c4e
https://github.com/AliceO2Group/O2DPG/commit/4acf74a6ff3121e10308edcdfabe357531abdde9
https://github.com/AliceO2Group/O2DPG/commit/d0b646e1ca82027503cdde58249dd99194137823

I will also have to think about how to deal with the specific `ConfigKeyValues` parameter classes
* QEDGenParam.h
* GenCosmicsParam.h

but I will come at a second moment with that.
Cheers,
Roberto

### CODE REVIEW

**Comment by preghenella** [comment]
hi @ktf @sawenzel 
why `clang-format` is apparently complaining about files that are being removed by this PR?

**Comment by ktf** [comment]
@preghenella https://github.com/AliceO2Group/AliceO2/pull/5642 should fix the formatting.

**Comment by preghenella** [comment]
`fullCI` check seems to break in `QC`, apparently unrelated to this PR.

File: prodtests/full_system_test.sh [hunk 1]
```diff
@@ -52,13 +52,6 @@ GLOBALDPLOPT="-b" #  --monitoring-backend no-op:// is currently removed due to h
 
 ulimit -n 4096 # Make sure we can open sufficiently many files
 [ $? == 0 ] || (echo Failed setting ulimit && exit 1)
-mkdir -p qed
-cd qed
-PbPbXSec="8."
-taskwrapper qedsim.log o2-sim --seed $O2SIMSEED -j $NJOBS -n$NEventsQED -m PIPE ITS MFT FT0 FV0 FDD -g extgen --configKeyValues '"GeneratorExternal.fileName=$O2_ROOT/share/Generators/external/QEDLoader.C;QEDGenParam.yMin=-7;QEDGenParam.yMax=7;QEDGenParam.ptMin=0.001;QEDGenParam.ptMax=1.;Diamond.width[2]=6."'
-QED2HAD=$(awk "BEGIN {printf \"%.2f\",`grep xSectionQED qedgenparam.ini | cut -d'=' -f 2`/$PbPbXSec}")
-echo "Obtained ratio of QED to hadronic x-sections = $QED2HAD" >> qedsim.log
-cd ..
```
Comment (sawenzel) [inline prodtests/full_system_test.sh:61]: The full system test needs QED contributions. I think this prevents moving QEDLoader away.

**Comment by shahor02** [comment]
@preghenella @sawenzel Cannot say that I like this idea, even keeping AEGIS outside of O2 costed a performance (e.g. impossible to account for O2 mag.field, use O2 track model etc.). Will you move the dependence of O2 on AEGIS to O2DPG?

For the FST, I don't see other way as to copy the needed macros or load O2DPG, both ways being ugly.

**Comment by preghenella** [comment]
hi @shahor02 
keeping AEGIS outside the O2 was planned since quite some time (we discussed this).

What is the problem with the magnetic field and the track model?
There will always be the possibility to load AEGIS and O2 packages simultaneously
`alienv enter AEGIS/latest,O2/latest`
and have the very same features that we have now.
There is no compiled code in the O2 that depends on AEGIS, as it should not be there.

Cheers,
Roberto

**Comment by davidrohr** [comment]
But if we need AEGIS in the full system test, couldn't we load it there?

**Comment by preghenella** [comment]
hi @davidrohr 
yes, you can (I think) load AEGIS environment and the O2DPG one (which is where the production macros are supposed to stay) for the fulls-sytem-test.
If we find it ugly to load the macros from the O2DPG, we can keep them in the O2, but given that these are macros for the production I would say that we should avoid keeping two copies and use the ones in the O2DPG

**Comment by davidrohr** [comment]
OK, then I would add AEGIS and O2DPG to the requirements of the o2-full-system-test.sh recipe.
This QC failure here is unrelated, but perhaps I would wait with merging  this PR until QC is fixed and until the o2-full-system-test in alidist is updated, such that the full CI reruns here, such that we see whether it works.

**Comment by preghenella** [comment]
hi @davidrohr 
sure, please wait. I will try to make sure that things do work as they shuld.
Thanks!

**Comment by sawenzel** [comment]
@davidrohr : But that would mean we can only execute the test as this recipe? 

My preference would be to just keep an internal (non-official copy) of QED macro for the purpose of this test. What argument really speaks against this?

**Comment by shahor02** [comment]
> keeping AEGIS outside the O2 was planned since quite some time (we discussed this).

Yes, that's why I did it in the way I did ...

> What is the problem with the magnetic field and the track model?

For the cosmic generator I needed to know in advance (before adding it to the stack) if the track generated at large R will traverse certain detectors (e.g. cylinders corresponding to ITS layers) requested from the command line. This would require access to alice track model and ITS geometry. So, I had to introduce a simplified track model + hardcoded cylindrical surfaces directly to the cosmic generator. And still the acceptance pre-simulation is approximate since it uses uniform field approximation...

**Comment by preghenella** [comment]
@sawenzel 
actually, keeping the QED macro is possibly not sufficient.
The QED library resides in the AEGIS package.

**Comment by sawenzel** [comment]
@preghenella : Well, AEGIS is a direct requirement of O2 because of QED. Actually, we have specific QED treatment in various places in O2, so I think it is even warranted to keep this dependency. 

Maybe I am missing something, but I don't see a danger of having a QED generator config inside of O2 (for testing) as well as having the official config in O2DPG (for PWG productions). What we could do is to move the O2-one outside of the Generators folder, so that there can be no ambiguity about this.

**Comment by davidrohr** [comment]
I agree with @sawenzel  on this. I mean for O2 without generators, we have the `o2-dataflow` default. The normal `o2` defaults have all the generators as dependencies, so why not the stuff needed for QED?

**Comment by preghenella** [comment]
we do not have generators as defaults in the `o2` and we will not have.
We only have `pythia` as a requirement, to provide external decayer that is needed at the geant simulation stage (so the `o2` code needs to build against them).

AEGIS was created as an external package to be used in conjunction with the `o2`.
The fact that it was a requirement so far was temporary and I strongly suggest we remove before we will not be able to do it anymore. It might start getting populated with user code very soon and it is prone to updates that will make the `o2` simulation core unstable.

Given that it is only testing that we are talking about here and that there is no limitation in the use of AEGIS interfaces even after it is removed from the `o2` requirements, I think it is time that we do this and make use of the simulation suite (o2, O2DPG and whatever other package that will provide generator) also in testing.

Cheers,
Roberto

**Pull request closed without merging**